pytools/test/test_persistent_dict.py

```diff
@@ -518,6 +518,51 @@ class TagClass2(Tag):
     assert keyb(TagClass2) != keyb(TagClass2())
 
 
+def test_dataclass_hashing():
+    keyb = KeyBuilder()
+
+    @dataclass
+    class MyDC:
+        name: str
+        value: int
+
+    assert keyb(MyDC("hi", 1)) == \
+        "2ba6363c3b98f1cc2209bd57388368b3efe3074e3764eee30fbcf15946efb802"
+
+    @dataclass
+    class MyDC2:
+        name: str
+        value: int
+
+    # Class types must be encoded in hash
+    assert keyb(MyDC2("hi", 1)) != keyb(MyDC("hi", 1))
+
+    try:
+        import attrs
+    except ImportError:
+        from warnings import warn
+        warn("attrs not installed, skipping rest of the test")
+        return
```

Comment:
Not sure what's better here, `warn` or `pytest.importorskip`.

Reply:
Break the `attrs` bit out into a separate test and use `importorskip`.

Reply:
Done in 4a1ac64